### PR TITLE
EVG-19977: Upgrade Go version, upgrade distro version, remove arch.

### PR DESCRIPTION
**Pull request by johndaniels**
Also upgrade MongoDB version used for testing to 6.

### CODE REVIEW

File: evergreen.yaml [hunk 1]
```diff
@@ -48,8 +48,20 @@ functions:
...
           DECOMPRESS: ${decompress}
+          MONGODB_URL: ${MONGODB_URL}
+          MONGODB_DECOMPRESS: ${MONGODB_DECOMPRESS|*decompress}
         working_dir: anser
         command: make get-mongodb
+    - command: subprocess.exec
+      type: setup
+      params:
+        command: make get-mongosh
+        include_expansions_in_env: ["MONGOSH_DOWNLOAD"]
```
Comment (ybrill) [inline evergreen.yaml:59]: this seems like this could have been left over

File: evergreen.yaml [hunk 2]
```diff
@@ -48,8 +48,20 @@ functions:
...
+        env:
+          DECOMPRESS: ${decompress}
```
Comment (ybrill) [inline evergreen.yaml:61]: This one seems unused

File: evergreen.yaml (hunk 1 above)
Comment (johndaniels) [reply evergreen.yaml]: Fixed

File: evergreen.yaml (hunk 2 above)
Comment (johndaniels) [reply evergreen.yaml]: Fixed

File: evergreen.yaml [hunk 3]
```diff
@@ -152,40 +161,36 @@ task_groups:
...
-    tasks: ["testGroup"]
-
   - name: lint
-    display_name: Lint (Arch Linux)
+    display_name: Lint
     expansions:
-      GOROOT: /opt/golang/go1.16
+      GOROOT: /opt/golang/go1.20
+      MONGODB_URL: https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu2204-6.0.6.tgz
+      MONGOSH_URL: https://downloads.mongodb.com/compass/mongosh-1.9.0-linux-x64.tgz
```
Comment (ybrill) [inline evergreen.yaml:169]: it seems like the lint tasks don't need mongo.

File: evergreen.yaml [hunk 4]
```diff
@@ -47,9 +47,18 @@ functions:
       params:
         include_expansions_in_env: ["MONGODB_URL"]
```
Comment (ybrill) [inline evergreen.yaml:48]: this seems superfluous in combination with passing it in `env`.

**Review by ybrill — APPROVED** [review]

**Pull request merged**